### DISCUSSION TITLE
improvement to columns, its all about the center

### DISCUSSION THREAD

**Discussion by Nebdir** [emilk/egui#1664] [Ideas]
I have struggled lately with centering a button in a horizontal row (see image below fig. 1).
After digging deeper and deeper into the layouts I've found that using columns is solving this halve way. 
By changing the alignment of the columns to center I would expect a layout like Figure 2.
Currently, the columns are aligned left with no option to change that (?). (see fig. 3)
Even further I think the limits of immediate UI's would prevent a layout like Figure 4
First of all, did I misread the code, or could this solve some problems? 
And if the function signature would change or another would be added are there downsides, I'm not currently aware of?
![layout](https://user-images.githubusercontent.com/6737610/169694966-2a926ce1-c8aa-4558-8d4e-64858f01b15d.png)

**Comment by emilk** [comment]
Columns Centered (fig 3) is the best you can do currently (until https://github.com/emilk/egui/issues/843), unless you know the sizes of the three widgets beforehand.
